### PR TITLE
Update to 2.11.5-SNAPSHOT

### PR DESCRIPTION
**Pull request by corneil**
Updates to snapshot.
Updates local test deployment scripts to register 2024.0.0/5.0.0 stream apps.

### CODE REVIEW

File: src/deploy/carvel/register-apps.sh [hunk 1]
```diff
@@ -43,10 +43,13 @@ else
     # unfortunately different in docker image names and registration link.
     BROKER_NAME=rabbitmq
 fi
-
 if [ "$STREAM_APPS_RT_VERSION" = "" ]; then
-    STREAM_URI="https://dataflow.spring.io/$BROKER_NAME-${TYPE}-latest"
-elif [[ "$STREAM_APPS_RT_VERSION" = *"SNAPSHOT"* ]]; then
+    # export STREAM_APPS_RT_VERSION=2022.0.0-SNAPSHOT
+    # export STREAM_APPS_RT_VERSION=2021.1.2 # release for Boot 2.x
+    export STREAM_APPS_RT_VERSION=2024.0.0
```
Comment (onobc) [inline src/deploy/carvel/register-apps.sh:49]: Main (`2.11.x`) should not ship w/ the 5.0 stream apps - this is for `3.0.x`

**Review by onobc — CHANGES_REQUESTED** [review]
Thanks for the updates @corneil . 

The changes for the snapshot update look good.

However, I don't think we want to update the stream apps version in `2.11.x` though as that would switch the default apps version in a patch release for the carvel packages. We are already updating to the latest apps in `main-3`.

Please submit a PR w/ only the snapshot updates.

**Review by cppwfs — CHANGES_REQUESTED** [review]

File: src/deploy/carvel/register-apps.sh (hunk 1 above)
Comment (corneil) [reply src/deploy/carvel/register-apps.sh:49]: Reverting

File: src/deploy/carvel/prepare-cluster.sh [hunk 2]
```diff
@@ -1,10 +1,36 @@
...
+SECRETGEN_CONTROLLER_VERSION=v0.17.3
 KAPP_CONTROLLER_VERSION=v0.50.0
+wget -q -O cert-manager-release.yml "https://github.com/cert-manager/cert-manager/releases/download/$CERT_MANAGER_VERSION/cert-manager.yaml"
+IMAGES=$(yq '.spec.template.spec.containers | .[] | .image' cert-manager-release.yml)
+for image in $IMAGES; do
+  if [[ "$image" != "---"* ]]; then
+    $K8S/load-image.sh "$image"
+  fi
+done
+#$K8S/load-image.sh "quay.io/jetstack/cert-manager-controller:$CERT_MANAGER_VERSION"
```
[the remark this answers is not in the document]
Comment (corneil) [reply src/deploy/carvel/prepare-cluster.sh:17]: Yes

File: src/deploy/carvel/register-apps.sh [hunk 3]
```diff
@@ -43,10 +43,12 @@ else
     # unfortunately different in docker image names and registration link.
     BROKER_NAME=rabbitmq
 fi
-
 if [ "$STREAM_APPS_RT_VERSION" = "" ]; then
-    STREAM_URI="https://dataflow.spring.io/$BROKER_NAME-${TYPE}-latest"
-elif [[ "$STREAM_APPS_RT_VERSION" = *"SNAPSHOT"* ]]; then
+    # export STREAM_APPS_RT_VERSION=2021.1.2 # release for Boot 2.x
+    export STREAM_APPS_RT_VERSION=2022.0.0
```
Comment (onobc) [inline src/deploy/carvel/register-apps.sh:48]: These are still SB3.x based apps. `2022.0.0 == 4.0 stream apps`

**Review by onobc — CHANGES_REQUESTED** [review]
Thanks for the updates @corneil . The stream apps are still not on the default version shipped w/ SCDF 2.11.x (`3.2.1 -> 2021.*`).

**Review by cppwfs — APPROVED** [review]
LGTM, for my requests.   Once @onobc marks it good, then merge.

File: src/deploy/carvel/register-apps.sh (hunk 3 above)
Comment (corneil) [reply src/deploy/carvel/register-apps.sh:48]: The docs specify that the default registered by the scripts are 2022.0.0

Comment (onobc) [reply src/deploy/carvel/register-apps.sh:48]: Where in the docs, we need to update that because I don't think that is accurate.

**Comment by corneil** [comment]
> Thanks for the updates @corneil . The stream apps are still not on the default version shipped w/ SCDF 2.11.x (`3.2.1 -> 2021.*`).

The docs specify that the registration scripts uses 2022.0.0

**Review by onobc — APPROVED** [review]

**Pull request merged**